airflow/www/views.py

```diff
@@ -710,24 +712,53 @@ def rendered(self):
             form=form,
             title=title,)
 
+    @expose('/get_log')
+    @login_required
+    @wwwutils.action_logging
+    def get_log(self):
+        """ The endpoint for fetching logs. """
+        logging_backend = cached_logging_backend()
+        if not logging_backend:
+            raise AirflowException("Logging backend not found.")
+
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        execution_date = request.args.get('execution_date')
+        dttm = dateutil.parser.parse(execution_date)
+        start_ts = request.args.get('start_ts')
+        end_ts = request.args.get('end_ts')
+
+        logs = logging_backend.get_logs(dag_id, task_id, execution_date,
+                                        start_ts=start_ts, end_ts=end_ts)
+        ts = start_ts if not logs else logs[-1].timestamp
```

Comment:
We need to keep track of timestamp of the last log line in order to fetch only the new portion.